WootricSDK/WootricSDK/WTRApiClient.m

```diff
@@ -150,68 +156,64 @@ - (void)updateExistingEndUser:(NSInteger)endUserID {
+  NSString *params = [NSString stringWithFormat:@"email=%@", escapedEmail];
     
-    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"%@/%@/end_users", _baseAPIURL, _apiVersion]];
-    NSString *escapedEmail = [self percentEscapeString:[_settings getEndUserEmailOrUnknown]];
-    NSString *params = [NSString stringWithFormat:@"email=%@", escapedEmail];
+  if (_settings.externalCreatedAt) {
+    params = [NSString stringWithFormat:@"%@&external_created_at=%ld", params, (long)[_settings.externalCreatedAt integerValue]];
+  }
     
-    if (_settings.externalCreatedAt) {
-        params = [NSString stringWithFormat:@"%@&external_created_at=%ld", params, (long)[_settings.externalCreatedAt integerValue]];
-    }
+  if (_settings.productName) {
+    params = [NSString stringWithFormat:@"%@&properties[product_name]=%@", params, _settings.productName];
+  }
     
-    if (_settings.productName) {
-        params = [NSString stringWithFormat:@"%@&properties[product_name]=%@", params, _settings.productName];
-    }
+  if (_settings.customProperties) {
+    NSString *parsedProperties = [WTRPropertiesParser parseToStringFromDictionary:_settings.customProperties];
+    params = [NSString stringWithFormat:@"%@%@", params, parsedProperties];
+  }
     
-    if (_settings.customProperties) {
-        NSString *parsedProperties = [WTRPropertiesParser parseToStringFromDictionary:_settings.customProperties];
-        params = [NSString stringWithFormat:@"%@%@", params, parsedProperties];
+  NSMutableURLRequest *urlRequest = [self requestWithURL:url HTTPMethod:@"POST" andHTTPBody:params];
+
+  NSURLSessionDataTask *dataTask = [_wootricSession dataTaskWithRequest:urlRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+    if (error) {
+      NSLog(@"WootricSDK (create end user): %@", error);
+    } else {
+      NSDictionary *responseJSON = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+      NSLog(@"WootricSDK (create end user): %@", responseJSON);
+      if (responseJSON) {
+        NSInteger endUserID = [responseJSON[@"id"] integerValue];
+        endUserWithID(endUserID);
+      }
     }
+  }];
     
-    NSMutableURLRequest *urlRequest = [self requestWithURL:url HTTPMethod:@"POST" andHTTPBody:params];
-    
-    NSURLSessionDataTask *dataTask = [_wootricSession dataTaskWithRequest:urlRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-        if (error) {
-            NSLog(@"WootricSDK (create end user): %@", error);
-        }
-        else {
-            NSDictionary *responseJSON = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
-            NSLog(@"WootricSDK (create end user): %@", responseJSON);
-            if (responseJSON) {
-                NSInteger endUserID = [responseJSON[@"id"] integerValue];
-                endUserWithID(endUserID);
-            }
-        }
-    }];
-    
-    [dataTask resume];
+  [dataTask resume];
```

Comment:
There's so much diffing everywhere, nothing changes here right ? use `git add patch` so you can see the chunks of code that you are going to add to your commit, in that way you can minimize meaningless chunks